catalog/video/catalog-mpegts-stream/src/main/java/com/connexta/alliance/video/stream/mpegts/CatalogRolloverAction.java

```diff
@@ -0,0 +1,358 @@
+        }
+    }
+
+    private UpdateRequest createUpdateRequest(String id, Metacard metacard) {
+        return new UpdateRequestImpl(id, metacard);
+    }
+
+    private void linkChildToParent(Metacard childMetacard) throws RolloverActionException {
+        setDerivedAttribute(childMetacard);
+
+        UpdateRequest updateChild = createUpdateRequest(childMetacard.getId(), childMetacard);
+
+        submitChildUpdateRequest(updateChild);
+    }
+
+    private void setDerivedAttribute(Metacard childMetacard) {
+        childMetacard.setAttribute(new AttributeImpl(Metacard.DERIVED, parentMetacard.getId()));
+    }
+
+    private CreateResponse submitStorageCreateRequest(CreateStorageRequest createRequest)
+            throws RolloverActionException {
+        try {
+            return catalogFramework.create(createRequest);
+        } catch (IngestException | SourceUnavailableException e) {
+            throw new RolloverActionException(String.format(
+                    "unable to submit storage create request to catalog framework: %s",
+                    createRequest), e);
+        }
+    }
+
+    private CreateStorageRequest createStorageRequest(ContentItem contentItem) {
+        return new CreateStorageRequestImpl(Collections.singletonList(contentItem),
+                new HashMap<>());
+    }
+
+    private void bindSecuritySubject() {
+        ThreadContext.bind(security.getSystemSubject());
+    }
+
+    private ContentItem createContentItem(MetacardImpl metacard, String fileName,
+            ByteSource byteSource) {
+        return new ContentItemImpl(byteSource, Constants.MPEGTS_MIME_TYPE, fileName, metacard);
+    }
+
+    private void createParentMetacard() throws RolloverActionException {
+        try {
+            createParentIfUnset();
+        } catch (MetacardCreationException | SourceUnavailableException | IngestException e) {
+            throw new RolloverActionException(String.format(
+                    "unable to create parent metacard: sourceUri=%s",
+                    streamProcessor.getStreamUri()), e);
+        }
+    }
+
+    private void createParentIfUnset()
+            throws MetacardCreationException, SourceUnavailableException, IngestException {
+        if (parentMetacard == null) {
+            MetacardImpl metacard = createInitialMetacard();
+
+            setParentResourceUri(metacard);
+            setParentTitle(metacard);
+            setParentContentType(metacard);
+
+            CreateRequest createRequest = new CreateRequestImpl(metacard);
+
+            submitParentCreateRequest(createRequest);
+
+        }
+    }
+
+    private void submitParentCreateRequest(CreateRequest createRequest)
+            throws IngestException, SourceUnavailableException {
+        catalogFramework.create(createRequest)
+                .getCreatedMetacards()
+                .forEach(createdMetacard -> {
+                    LOGGER.info("created parent metacard: metacard={}", createdMetacard.getId());
```

Comment:
This will be better as trace or debug.  With a large amount of created metacards, this will pollute the log with an enormous amount of information and perhaps cause some performance issues.